src/runtime/Types/MetaType.cs

```diff
@@ -79,49 +82,59 @@ public static NewReference tp_new(BorrowedReference tp, BorrowedReference args,
             BorrowedReference bases = Runtime.PyTuple_GetItem(args, 1);
             BorrowedReference dict = Runtime.PyTuple_GetItem(args, 2);
 
-            // We do not support multiple inheritance, so the bases argument
-            // should be a 1-item tuple containing the type we are subtyping.
-            // That type must itself have a managed implementation. We check
-            // that by making sure its metatype is the CLR metatype.
+            // Extract interface types and base class types.
+            var interfaces = new List<Type>();
 
-            if (Runtime.PyTuple_Size(bases) != 1)
-            {
-                return Exceptions.RaiseTypeError("cannot use multiple inheritance with managed classes");
-            }
+            // More than one base type case be declared, but an exception will be thrown
+            // if more than one is a class/not an interface.
+            var baseTypes = new List<ClassBase>();
 
-            BorrowedReference base_type = Runtime.PyTuple_GetItem(bases, 0);
-            BorrowedReference mt = Runtime.PyObject_TYPE(base_type);
+            var baseClassCount = Runtime.PyTuple_Size(bases);
 
-            if (!(mt == PyCLRMetaType || mt == Runtime.PyTypeType))
+            for (nint i = 0; i < baseClassCount; i++)
             {
-                return Exceptions.RaiseTypeError("invalid metatype");
+                var baseTypeIt = Runtime.PyTuple_GetItem(bases, (int)i);
+
+                if (GetManagedObject(baseTypeIt) is ClassBase classBaseIt)
+                {
+                    if (classBaseIt.type.Valid && classBaseIt.type.Value.IsInterface)
```

Comment:
Invalid `ClassBase` instances should also raise an error regardless of anything else.

Reply:
got it, adding exceptions.